src/components/ContactsAdd.js

```diff
@@ -1,39 +1,156 @@
-import { useState } from "react"
-import { useNavigate } from "react-router-dom";
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 
-function ContactsAdd(props) {
+function ContactsAdd({ contacts, setContacts }) {
+	const [addContactFormEmpty, setAddContactFormEmpty] = useState({
+		firstName: '',
+		lastName: '',
+		street: '',
+		city: '',
+		email: '',
+		linkedIn: '',
+		twitter: '',
+	});
 
-  // setContacts and contacts must be passed as props
-  // to this component so new contacts can be added to the
-  // state
-  const { setContacts, contacts } = props
+	const [addContactForm, setAddContactForm] = useState(addContactFormEmpty);
 
-  //TODO: Implement controlled form
-  //send POST to json server on form submit
+	let navigate = useNavigate();
 
-  return (
-    <form className="form-stack contact-form">
-      <h2>Create Contact</h2>
+	const onFirstNameChange = (event) => {
+		setAddContactForm({ ...addContactForm, firstName: event.target.value });
+	};
 
-      <label htmlFor="firstName">First Name</label>
-      <input id="firstName" name="firstName" type="text" required />
+	const onLastNameChange = (event) => {
```

Comment:
Theres a way to simplify these functions with event.target.name: event.target.value. Might have made the code a bit drier